src/Libraries/Microsoft.AspNetCore.HeaderParsing/Metric.cs

```diff
@@ -8,9 +8,9 @@ namespace Microsoft.AspNetCore.HeaderParsing;
 
 internal static partial class Metric
 {
-    [Counter("HeaderName", "Kind", Name = @"HeaderParsing.ParsingErrors")]
+    [Counter("header.name", "error.type", Name = "aspnetcore.header_parsing.parse_errors")]
```

Comment:
I don't know if we're still at the stage to discuss API, but making tag names the param array here feels kind of weird. I could imagine someone entering `[Counter("my.cool.counter.name")] ` and expecting a counter with that name but instead getting a counter with no name a tag that name instead.

IMO `CounterAttribute` should have a ctor that takes the name. It's the most significant and required piece of information for a counter. Then specify a type or array of strings as tags as optional parameter, e.g.

```cs
[Counter("my.cool.counter.name", Tags = [ "one", "two", "etc" ])]
```

Reply:
Added a comment to the API review PR: https://github.com/dotnet/extensions/pull/4373/files#r1351364773

Reply:
Thanks for raising that. It's out of scope of this PR, so I'm resolving this thread. Feel free to re-open it.